### DISCUSSION TITLE
How can i get id of last sended message by bot?

### DISCUSSION THREAD

**Discussion by Adevald** [nutgram/nutgram#496] [Q&A] [answered]
As i understand, php not so async so i cant just take Promise from `sendMessage` function like i should do it in JS.

As an example this feature can help me to manage edit last message in bot to simulate UI with cb keyboard. Or remove from group some repeating info messages sended by bot.

**Comment by Lukasss93** [comment, marked as answer]
The `sendMessage` method returns a [Message](https://github.com/nutgram/nutgram/blob/master/src/Telegram/Types/Message/Message.php) object. Just use this object to get what you want:
```php
$message = $bot->sendMessage('Hello!');

// option 1
$bot->deleteMessage($message->chat->id, $message->message_id);

// option 2
$message->delete();
```

**Reply by bogdasari4** [reply]
How can I get **message_id** using `sendMediaGroup()`?

**Reply by Lukasss93** [reply]
https://core.telegram.org/bots/api#sendmediagroup

sendMediaGroup returns an array of Messages:

$messages = $bot->sendMediaGroup(your-code);

$messages[0]->message_id
$messages[1]->message_id
....
